addons/website/static/src/snippets/s_dynamic_snippet/options.js

```diff
@@ -47,15 +47,6 @@ const dynamicSnippetOptions = options.Class.extend({
         if (params.attributeName === 'templateKey' && previewMode === false) {
             this._templateUpdated(widgetValue, params.activeValue);
         }
-        if (params.attributeName === 'numberOfRecords' && previewMode === false) {
-            const dataSet = this.$target.get(0).dataset;
-            if (dataSet.numberOfElements > dataSet.numberOfRecords) {
-                dataSet.numberOfElements = dataSet.numberOfRecords;
-            }
-            if (dataSet.numberOfElementsSmallDevices > dataSet.numberOfRecords) {
-                dataSet.numberOfElementsSmallDevices = dataSet.numberOfRecords;
-            }
```

Comment:
The problem of the missing `parseInt` in number comparisons should be done in a stable version (15.0 ?)